pkg/monitor/api.go

```diff
@@ -182,6 +193,117 @@ func startServerMonitoring(ctx context.Context, m *Monitor, clusterConfig *rest.
 	return nil
 }
 
+func startNetworkMonitoring(ctx context.Context, m *Monitor, clusterConfig *rest.Config, timeout time.Duration, resourceLocator, url string, disableConnectionReuse bool) error {
+	kubeTransportConfig, err := clusterConfig.TransportConfig()
+	if err != nil {
+		return err
+	}
+	tlsConfig, err := transport.TLSConfigFor(kubeTransportConfig)
+	if err != nil {
+		return err
+	}
+	var httpTransport *http.Transport
+
+	if disableConnectionReuse {
+		httpTransport = &http.Transport{
+			Dial: (&net.Dialer{
+				Timeout:   timeout,
+				KeepAlive: -1, // this looks unnecessary to me, but it was set in other code.
+			}).Dial,
+			TLSClientConfig:     tlsConfig,
+			TLSHandshakeTimeout: timeout,
+			DisableKeepAlives:   true, // this prevents connections from being reused
+			IdleConnTimeout:     timeout,
+		}
+	} else {
+		httpTransport = &http.Transport{
+			Dial: (&net.Dialer{
+				Timeout: timeout,
+			}).Dial,
+			TLSClientConfig:     tlsConfig,
+			TLSHandshakeTimeout: timeout,
+			IdleConnTimeout:     timeout,
+		}
+	}
+
+	roundTripper := http.RoundTripper(httpTransport)
+	if kubeTransportConfig.HasTokenAuth() {
+		roundTripper, err = transport.NewBearerAuthWithRefreshRoundTripper(kubeTransportConfig.BearerToken, kubeTransportConfig.BearerTokenFile, httpTransport)
+		if err != nil {
+			return err
+		}
+
+	}
+
+	httpClient := http.Client{
+		Transport: roundTripper,
+	}
+
+	go NewSampler(m, time.Second, func(previous bool) (condition *monitorapi.Condition, next bool) {
+		resp, err := httpClient.Get(clusterConfig.Host + url)
+
+		if err == nil && resp != nil && resp.Body != nil {
+			defer resp.Body.Close()
+		}
+
+		switch {
+		case err == nil && !previous:
+			condition = &monitorapi.Condition{
+				Level:   monitorapi.Info,
+				Locator: resourceLocator,
+				Message: fmt.Sprintf("%s started responding to network requests", resourceLocator),
+			}
+		case err != nil && IsNetworkError(err) && previous:
+			condition = &monitorapi.Condition{
+				Level:   monitorapi.Error,
+				Locator: resourceLocator,
+				Message: fmt.Sprintf("%s started failing: %v", resourceLocator, err),
+			}
+		}
+		return condition, IsNetworkError(err)
+	}).WhenFailing(ctx, &monitorapi.Condition{
+		Level:   monitorapi.Error,
+		Locator: resourceLocator,
+		Message: fmt.Sprintf("%s is not responding to network requests", resourceLocator),
+	})
+
+	return nil
+}
+
+var apiMachineryErrors = []string{
+	"connection refused", // TODO: Likely a networking issue, but require more analysis to tell
+	"TLS handshake timeout",
+	"EOF", // could be ungraceful termination
+}
+
+var networkErrors = []string{
+	"i/o timeout",
+	"connection reset by peer",
```

Comment:
meanwhile `connection refused` happens when we are starting a connection, this means that the connection was already established and it was working, however, something resets it.
I saw this a lot on loadbalancers and proxies, i.e, the upstream server hangs or ir not available.
It also can be sent by the server if it is overloaded or not able to process a request ... 
This is tricky to categorize ... but the fact that a connection has already been established makes me think that is more an apiMachinery error than a network error